src/ecmascript_simd.js

```diff
@@ -107,17 +103,19 @@ function clamp(a, min, max) {
 // SIMD implementation functions
 
 function simdCoerceIndex(index) {
-    index = +index;
-    if (index != Math.floor(index))
-        throw new RangeError("SIMD index must be an integer");
-    return index;
+  index = +index;
+  if (index != Math.floor(index))
+    throw new RangeError("SIMD index must be an integer");
+  return index;
 }
 
-function simdCheckLaneIndex(index, lanes) {
-  if (!isInt32(index))
-    throw new TypeError('Lane index must be an int32');
-  if (index < 0 || index >= lanes)
+function simdCoerceLaneIndex(index, lanes) {
+  index = +index;
+  if (!(index >= 0 && index < lanes))
     throw new RangeError('Lane index must be in bounds');
+  if (index != Math.floor(index))
+    throw new RangeError('Lane index must be an integer');
+  return index;
 }
 
```

Comment:
Maybe implement simdCoerceLaneIndex() using simdCoerceIndex(), just to avoid a little code duplication.  It shouldn't matter that the index != Math.floor() check fails before the bounds check, I think.